src/cursor.rs

```diff
@@ -473,7 +482,24 @@ impl Cursor {
             self.batch_size
         };
 
-        self.next_n(batch_size)
+        // If `usize` is at least as wide as `i32`, then all non-negative values
+        // of `i32` are guaranteed to fit into a `usize`. Otherwise `usize::MAX`
+        // fits into an `i32`.
+        if
+            batch_size < 0
+            ||
+            (
+                size_of::<usize>() < size_of::<i32>()
```

Comment:
If we're going to this check like above, I think it might make sense to do it statically (once) rather than repeatedly at run-time; making a `USIZE_SMALLER_THAN_32_BITS` constant with `lazy_static` might make sense.

Reply:
Do you mean that from a performance point of view? If so, I think that might actually be a tiny bit worse – I can check but I'm pretty sure this will be optimized out at compile time exactly because it's known statically. A `lazy_static` would however require a `bool` worth of memory to be allocated dynamically, and it would add locking on each access (deref) – unless the compiler is smart enough to optimize even the `lazy_static`, at which point it would likely do away with the current runtime check too anyway.

Reply:
Fair enough; we can leave this as-is